### PR TITLE
Make sure ctype calls use unsigned chars.

### PR DESCRIPTION
**Pull request by JC3**
Addresses https://github.com/assimp/assimp/issues/3867 and then some. See discussion there. 

This also includes the change from #3868, which I've closed.

### CODE REVIEW

**Comment by JC3** [comment]
Also submitted https://github.com/google/googletest/pull/3393.

**Review by kimkulling — APPROVED** [review]

**Comment by kimkulling** [comment]
Merged, thanks a lot for your contribution.

File: code/AssetLib/Collada/ColladaParser.cpp [hunk 1]
```diff
@@ -234,7 +234,7 @@ void ColladaParser::UriDecodePath(aiString &ss) {
 #if defined(_MSC_VER)
     if (ss.data[0] == '/' && isalpha((unsigned char)ss.data[1]) && ss.data[2] == ':') {
 #else
-    if (ss.data[0] == '/' && isalpha(ss.data[1]) && ss.data[2] == ':') {
+    if (ss.data[0] == '/' && isalpha((unsigned char)ss.data[1]) && ss.data[2] == ':') {
 #endif
```
Comment (krishty) [inline code/AssetLib/Collada/ColladaParser.cpp:238]: Can we remove that `#if` now altogether?

File: code/AssetLib/X/XFileImporter.cpp [hunk 2]
```diff
@@ -667,8 +667,8 @@ void XFileImporter::ConvertMaterials( aiScene* pScene, std::vector<XFile::Materi
 
                 // convert to lower case for easier comparison
                 for ( unsigned int c = 0; c < sz.length(); ++c ) {
-                    if ( isalpha( sz[ c ] ) ) {
-                        sz[ c ] = (char) tolower( sz[ c ] );
+                    if ( isalpha( (unsigned char) sz[ c ] ) ) {
```
Comment (krishty) [inline code/AssetLib/X/XFileImporter.cpp:670]: `isalpha()` is useless here if I read `tolower()` specification right

File: code/AssetLib/Collada/ColladaParser.cpp (hunk 1 above)
Comment (JC3) [reply code/AssetLib/Collada/ColladaParser.cpp:238]: Yes

**Pull request merged**